tests/phpunit/integration/Modules/Tag_ManagerTest.php

```diff
@@ -419,6 +493,63 @@ public function test_sanitize_container_name( $input, $expected ) {
 		);
 	}
 
+	/**
+	 * @dataProvider data_ads_measurement_data
+	 */
+	public function test_check_ads_measurement_connection( $tags, $expected_result ) {
+		$account_id            = '123456789';
+		$container_id          = 'GTM-987654321';
+		$internal_container_id = 'GTM-987654321';
+
+		$this->tagmanager->get_settings()->merge(
+			array(
+				'containerID'         => $container_id,
+				'accountID'           => $account_id,
+				'internalContainerID' => $internal_container_id,
+			)
+		);
+
+		$this->setup_user_authentication( 'valid-auth-token' );
+
+		$this->authentication->get_oauth_client()->set_granted_scopes(
+			$this->tagmanager->get_scopes()
+		);
+
+		FakeHttp::fake_google_http_handler(
+			$this->tagmanager->get_client(),
+			function ( Request $request ) use ( $account_id, $container_id, $tags ) {
+				if ( false !== strpos( $request->getUri(), "/accounts/{$account_id}/containers/{$container_id}/versions:live" ) ) {
```

Comment:
The test with tag present was passing incorrectly because the URL was constructed using the `container_id` when it uses the internal ID instead. These had the same value defined above which is why it seemed to work.